lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/ServiceAndContextConfigurationProvider.cs

```diff
@@ -166,9 +175,13 @@ internal ContextAndServiceConfiguration GetDataLoadingConfigurationForEvaluatorB
 
             var serviceConf =
                 TangFactory.GetTang()
-                    .NewConfigurationBuilder(ServiceConfiguration.ConfigurationModule.Build(),
+                    .NewConfigurationBuilder(
+                        ServiceConfiguration.ConfigurationModule
+                        .Set(ServiceConfiguration.Services, GenericType<TaskStateService>.Class)
```

Comment:
ContextHandler is in particular to do something when context starts. It's OnNext() will be called when context starts. Service is an object that we would like to inject to the context level.
 
We had long discussion about the Service on context, mainly is about the name of the service. We agreed it is essentially any object, not a service.